pkg/diagnostics/collectors.go

```diff
@@ -380,6 +380,56 @@ func (c *collectorFactory) crdCollector(crdType string) *Collect {
 	}
 }
 
+// APIServerCollector collect connection info when running a pod on an existing cluster
+func (c *collectorFactory) APIServerCollector(controlPlaneIP string) []*Collect {
+	var collectors []*Collect
+	collectors = append(collectors, c.generateRunPodsCollector(controlPlaneIP)...)
+	// collectors = append(collectors, c.pingLogCollectors()...)
+	return collectors
+}
+
+func (c *collectorFactory) generateRunPodsCollector(controlPlaneIP string) []*Collect {
+	return c.runPodsCollectors(controlPlaneIP)
+}
+
+func (c *collectorFactory) runPodsCollectors(controlPlaneIP string) []*Collect {
+	tempIPRequest := "yum install -y nmap && for port in 6443 22; do nc -z -v -w5 " + controlPlaneIP + " $port; done"
```

Comment:
rather than run `yum install` here, we can use `yum` to install `nmap` directly on the base image that is used for the Diagnostic Collector image.

Check out the dockerfile here:

https://github.com/aws/eks-anywhere-build-tooling/blob/a27f3bd332d5374b4519ed48aedc2144c107996c/projects/aws/eks-anywhere/docker/linux/eksa-diagnostic-collector/Dockerfile

Currently it only has tar installed but we could also install NMAP as well; it seems like it would be generically useful.

Let me know if you need any assistance here.

Reply:
I agree with this approach.

Reply:
https://github.com/aws/eks-anywhere-build-tooling/pull/1077